app/main/posts/modify/post-media.directive.js

```diff
@@ -16,15 +16,27 @@ function (
     return {
         restrict: 'E',
         replace: true,
-        require: '^^form',
+        require: ['^^form', 'ngModel'],
         scope: {
             mediaId: '=',
             media: '=',
             name: '@',
             mediaHasCaption: '='
         },
         template: require('./media.html'),
-        link: function ($scope, element, attr, formCtrl) {
+        link: function ($scope, element, attr, requiredAttributes) {
+            var form = requiredAttributes[0];
+            var ngModelForm = requiredAttributes[1];
+
+            ngModelForm.$render = renderViewValue;
+
+            function renderViewValue() {
+                if (ngModelForm.$viewValue === 'changed') {
```

Comment:
You shouldn't need to do this at all. But you need to use `ngModelForm.$setViewValue();` to save the value